backend/layers/processing/process_validate_atac.py

```diff
@@ -0,0 +1,224 @@
+import hashlib
+
+from backend.common.utils.corpora_constants import CorporaConstants
+from backend.layers.business.business_interface import BusinessLogicInterface
+from backend.layers.common.entities import (
+    ARTIFACT_TO_EXTENSION,
+    CollectionVersionId,
+    DatasetArtifactId,
+    DatasetArtifactType,
+    DatasetConversionStatus,
+    DatasetStatusKey,
+    DatasetValidationStatus,
+    DatasetVersionId,
+)
+from backend.layers.common.ingestion_manifest import IngestionManifest
+from backend.layers.processing.exceptions import ConversionFailed, ValidationAtacFailed
+from backend.layers.processing.process_logic import ProcessingLogic
+from backend.layers.thirdparty.s3_provider_interface import S3ProviderInterface
+from backend.layers.thirdparty.schema_validator_provider import SchemaValidatorProviderInterface
+from backend.layers.thirdparty.uri_provider import UriProviderInterface
+
+
+class ProcessValidateATAC(ProcessingLogic):
+    def __init__(
+        self,
+        business_logic: BusinessLogicInterface,
+        uri_provider: UriProviderInterface,
+        s3_provider: S3ProviderInterface,
+        schema_validator: SchemaValidatorProviderInterface,
+    ) -> None:
+        super().__init__()
+        self.business_logic = business_logic
+        self.uri_provider = uri_provider
+        self.s3_provider = s3_provider
+        self.schema_validator = schema_validator
+
+    def create_atac_artifact(
+        self,
+        file_name: str,
+        artifact_type: DatasetArtifactType,
+        dataset_version_id: DatasetVersionId,
+        processing_status_key: DatasetStatusKey,
+        datasets_bucket: str,
+        fragment_artifact_id: DatasetArtifactId = None,
+    ) -> DatasetArtifactId:
+        """
+        Uploads the file to S3 and updates the database with the artifact using the artifact_id as the prefix for the
+        key.
+        :param file_name: the local file to upload
+        :param artifact_type: the type of artifact to upload
+        :param dataset_version_id: the dataset version id
+        :param processing_status_key: the key to update the processing status
+        :param datasets_bucket: the bucket to upload the dataset to
+        :param fragment_artifact_id: the artifact id of the fragment file to be use in the fragment index file.
+        :return:
+        """
+        self.update_processing_status(dataset_version_id, processing_status_key, DatasetConversionStatus.UPLOADING)
+        try:
+            artifact_id = self.business_logic.add_dataset_artifact(dataset_version_id, artifact_type, "dummy")
```

Comment:
why are we passing in a dummy value?

Reply:
I'd rather add a new function that generates an artifact ID, and create (or reuse if there is one) a function that lets us insert with our preferred artifact ID instead.